sync-core/src/main/java/com/cloudant/mazha/CouchClient.java

```diff
@@ -542,34 +614,35 @@ private String createBulkSerializedDocsPayload(List<String> serializedDocs) {
                 res.put(e.getKey(), e.getValue().missing);
             }
             return res;
+        } catch (IOException ioe) {
+            // TODO
+            return null;
         } finally {
             closeQuietly(is);
         }
     }
 
-    public Response putMultipart(MultipartAttachmentWriter mpw) {
+    public Response putMultipart(final MultipartAttachmentWriter mpw) {
         Map<String, Object> options = new HashMap<String, Object>();
         options.put("new_edits", "false");
         URI uri = this.uriHelper.documentUri(mpw.getId(), options);
         HashMap<String, String> headers = new HashMap<String, String>();
         String contentType = "multipart/related;boundary=" + mpw.getBoundary();
-        InputStream is = this.httpClient.putStream(uri, contentType, mpw, mpw.getContentLength());
-        return jsonHelper.fromJson(new InputStreamReader(is), Response.class);
+        HttpConnection connection = Http.PUT(contentType, uri);
+        connection.setInput(mpw, mpw.getContentLength());
+        try {
+            InputStream is = connection.executeToInputStream();
+            return jsonHelper.fromJson(new InputStreamReader(is), Response.class);
```

Comment:
Would it be useful to have a method on HttpConnection, `responseAsClass(Response.class)`?